projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py

```diff
@@ -96,246 +238,177 @@ def vdk_configure(self, config_builder: ConfigurationBuilder):
 
         config_builder.add(
             key=STRUCTLOG_CONSOLE_LOG_PATTERN,
-            default_value="",
-            description="Custom format string for console logging. Leave empty for default format.",
+            default_value=None,
+            description=(
+                "Custom format string for console logging, applied only when`logging_format` is 'console'. "
+                "Overrides `logging_metadata`. Note: For config.ini, %-signs should be escaped by doubling, "
+                "e.g. %(asctime)s should become %%(asctime)s"
+            ),
         )
 
         config_builder.add(
             key=STRUCTLOG_LOGGING_FORMAT_KEY,
-            default_value=STRUCTLOG_LOGGING_FORMAT_DEFAULT,
+            default_value=None,
             description=(
                 f"Controls the logging output format. Possible values: {STRUCTLOG_LOGGING_FORMAT_POSSIBLE_VALUES}"
             ),
         )
 
         config_builder.add(
             key=SYSLOG_HOST_KEY,
-            default_value=DEFAULT_SYSLOG_HOST,
+            default_value=None,
```

Comment:
It's better to leave the default values here because 

A) They are used to infer the type and convert to correct type (though default type is str) 
B) They show up in `vdk config-help` and in other help and configuration documentation we will generate 

It seems get using Configuration.get_value(key) ([link to code](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/core/config.py#L73)) there's no way to know if value is set or it's the default passed here. 

But we need to solve that without breaking current good practices above. 

I don't have many good ideas. 
```
    def get_value(self, key: ConfigKey, fallback_value: ConfigValue) -> ConfigValue:
                   key = _normalize_config_key(key)
               
        default_value = fallback_value if fallback_value else self.__config_key_to_default_value.get(key)
        value = self.__config_key_to_value.get(key, default_value)
        return value
```
 
 This would allow you do `configuration.get_value(key, presets.getdefault(preset_key, {}).get(key, None))` 
 
 I am not sure if this is the best design. It may work, but it can be a bit complex and may not be the most straightforward approach. 
 
 Alterantively we can introduce is_value_set(key) function? 
 
 (If you do any changes in vdk-core, it must be in separate PR and merged before this one is merged)
 
  What do you think? 
  
 There's need of more thinking here. This is not that straightforward task.

Reply:
I think this should be a separate ticket.

Reply:
If we merge this PR as it is , I think it will leave technical debt as it would introduce some inconsistencies in how configurations are handled (above points A) And B) ) . 

And consistency is pretty crucial for maintainability, usability, and avoiding potential bugs.
First we should agree on that. 

We could merge it if we have a clear plan for revisiting it (aka story/issue) within the near future ( next sprint).

Reply:
Ok, let's do this first instead https://github.com/vmware/versatile-data-kit/pull/3076

I have some capacity left over for this sprint, so it should be fine.